x/programs/rust/wasmlanche_sdk/src/host/mod.rs

```diff
@@ -97,35 +102,3 @@ pub fn host_program_invoke(
         )
     }
 }
-
-/* memory functions ------------------------------------------- */
-// https://radu-matei.com/blog/practical-guide-to-wasm-memory/
-
-/// Allocate memory into the module's linear memory
-/// and return the offset to the start of the block.
-#[no_mangle]
-pub fn alloc(len: usize) -> *mut u8 {
-    // create a new mutable buffer with capacity `len`
-    let mut buf = Vec::with_capacity(len);
-    // take a mutable pointer to the buffer
-    let ptr = buf.as_mut_ptr();
-    // take ownership of the memory block and
-    // ensure that its destructor is not
-    // called when the object goes out of scope
-    // at the end of the function
-    std::mem::forget(buf);
-    // return the pointer so the runtime
-    // can write data at this offset
-    ptr
-}
-
-/// # Safety
-/// `ptr` must be a pointer to a block of memory.
-///
-/// deallocates the memory block at `ptr` with a given `capacity`.
-#[no_mangle]
-pub unsafe fn dealloc(ptr: *mut u8, capacity: usize) {
-    // always deallocate the full capacity, initialize vs uninitialized memory is irrelevant here
-    let data = Vec::from_raw_parts(ptr, capacity, capacity);
-    std::mem::drop(data);
-}
```

Comment:
Clippy didn't like the usage of `#[no_mangle]` here. It doesn't make sense to have the `#[no_mangle]` attribute if the function isn't being exported with `extern "C" fn // ...`. 

So instead of exposing a function that isn't being called from the wasm-host, I figured it's probably better to delete these. Let me know if you (the reviewer) thinks otherwise.

Reply:
i think you need #[no_mange] when dealing with wazero and the rust compiler. Otherwise, the compiler can you change the function signature which causes an error in the host.

Reply:
you should also have `extern "C"` in that case (at least that's what clippy says)